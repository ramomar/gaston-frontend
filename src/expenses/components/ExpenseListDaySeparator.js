import React from 'react';
import PropTypes from 'prop-types';
import { Text, Box } from 'grommet';
import { DateTime } from 'luxon';

function ExpenseListDaySeparator(props) {
  const formattedDay = props.day.toFormat(`d 'de' MMMM`);

  return (
    <Box
      height='xxsmall'
      margin={{ bottom: 'medium' }}
      pad={{ vertical: 'large' }}
      direction='row'
      align='center'
      justify='between'
      background='white'
      style={{ position: 'sticky', top: 0 }}>
      <Text weight='bold'>{formattedDay}</Text>
      <Box align='end'>
        <Text color='red' weight='bold' size='small'>-${props.amountForDay.toFixed(2)}</Text>
      </Box>
    </Box>
  );
}

ExpenseListDaySeparator.propTypes = {
  day: PropTypes.instanceOf(DateTime).isRequired,
  amountForDay: PropTypes.number.isRequired
};

export default ExpenseListDaySeparator;
