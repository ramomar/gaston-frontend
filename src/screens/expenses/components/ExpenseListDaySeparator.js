import React from 'react';
import { Text, Box } from 'grommet';

function ExpenseListDaySeparator({ day, amountForDay }) {
  const formattedDay = day.toFormat(`d 'de' MMMM`);

  return (
    <Box
      pad='medium'
      margin={{ top: 'medium' }}
      pad='large'
      direction='row'
      align='center'
      justify='between'
      background='white'
      style={{ position: 'sticky', top: 0 }}>
      <Text weight='bold'>{formattedDay}</Text>
      <Box align='end'>
        <Text color='red' weight='bold' size='small'>-${amountForDay.toFixed(2)}</Text>
      </Box>
    </Box>
  );
}

export default ExpenseListDaySeparator;
