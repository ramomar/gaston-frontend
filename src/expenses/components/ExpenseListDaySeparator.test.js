import React from 'react';
import renderer from 'react-test-renderer';
import ExpenseListDaySeparator from './ExpenseListDaySeparator';
import { DateTime } from 'luxon';

it('renders correctly', () => {
  const component =
    <ExpenseListDaySeparator
      day={DateTime.fromISO('2019-12-30T02:58:24.948Z')}
      amountForDay={100.50} />;

  const tree = renderer.create(component).toJSON();

  expect(tree).toMatchSnapshot();
});
