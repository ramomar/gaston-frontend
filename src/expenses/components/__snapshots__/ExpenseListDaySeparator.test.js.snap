// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 jNYyVh"
  style={
    Object {
      "position": "sticky",
      "top": 0,
    }
  }
>
  <span
    className="StyledText-sc-1sadyjn-0 bnqisW"
  >
    29 de December
  </span>
  <div
    className="StyledBox-sc-13pk1d4-0 dRsJGz"
  >
    <span
      className="StyledText-sc-1sadyjn-0 hLzlgx"
      size="small"
    >
      -$
      100.50
    </span>
  </div>
</div>
`;
